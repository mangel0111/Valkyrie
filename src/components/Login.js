import React from 'react';
//import base from '../base';

class Login extends React.Component {
   
    render() {
        return (
            <nav className="login">
                <h2>Valkyrie!</h2>
                <p>Sign in to use yourApp tool!</p>
                <div>
                    <button className="github" onClick={() => this.authenticate('github')}>Log In with Github</button>
                    <button className="facebook" onClick={() => this.authenticate('facebook')} >Log In with Facebook</button>
                    <button className="twitter" onClick={() => this.authenticate('twitter')} >Log In with Twitter</button>
                    <button className="google" onClick={() => this.authenticate('google')} >Log In with Google</button>
                </div>
                <p className='footnote'>All rights reserved (R)</p>
            </nav>
        )
    }
}

export default Login;
