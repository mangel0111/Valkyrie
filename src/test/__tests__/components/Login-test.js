import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../../../components/Login';
//import { mockComponent } from '../../utils/testUtils';

jest.mock('react-google-login',  () => 'googleLogin');

it('renders correctly', () => {
    const tree = renderer.create(
        <Login />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
