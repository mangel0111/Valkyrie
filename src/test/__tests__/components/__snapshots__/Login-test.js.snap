// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="login"
>
  <h2>
    Valkyrie!
  </h2>
  <p>
    Sign in to use yourApp tool!
  </p>
  <div>
    <googleLogin
      buttonText="Log In with Google"
      className="google"
      clientId="576488167011-22sf5nn422hp65mq81aj93h0hfaqb1vu.apps.googleusercontent.com"
      onFailure={[Function]}
      onSuccess={[Function]}
      scope="https://www.googleapis.com/auth/calendar.readonly profile"
    />
  </div>
  <p
    className="footnote"
  >
    All rights reserved (R)
  </p>
</nav>
`;
